internal/framework/provider/config.go

```diff
@@ -0,0 +1,103 @@
+package provider
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/mongodb-forks/digest"
+	"github.com/mongodb/terraform-provider-mongodbatlas/version"
+	"github.com/spf13/cast"
+	matlasClient "go.mongodb.org/atlas/mongodbatlas"
+	realmAuth "go.mongodb.org/realm/auth"
+
+	"go.mongodb.org/realm/realm"
+)
+
+type Config struct {
+	PublicKey    string
+	PrivateKey   string
+	BaseURL      string
+	RealmBaseURL string
+	AssumeRole   *AssumeRole
+}
+
+type AssumeRole struct {
+	RoleARN           string
+	Duration          time.Duration
+	ExternalID        string
+	Policy            string
+	PolicyARNs        []string
+	SessionName       string
+	SourceIdentity    string
+	Tags              map[string]string
+	TransitiveTagKeys []string
+}
+
+// MongoDBClient client
+type MongoDBClient struct {
+	Atlas  *matlasClient.Client
+	Config *Config
+}
+
+var ua = "terraform-provider-mongodbatlas/" + version.ProviderVersion
```

Comment:
Let's make this similar to other tools

```suggestion
const ToolName       = "terraform-provider-mongodbatlas"
var UserAgent      = fmt.Sprintf("%s/%s (%s;%s)", ToolName, version.ProviderVersion, runtime.GOOS, runtime.GOARCH))
```